src/screens/Customers/CustomerMain.tsx

```diff
@@ -1,228 +1,231 @@
-  const [allCustomersTrie, setAllCustomersTrie] = useState<TrieTree<CustomerRecord>>(new TrieTree('name'));
-  const [sortBy, setSortBy] = useState<SortBy>(SortBy.NAME);
-  const [filterBy, setFilterBy] = useState<FilterBy>(FilterBy.ACTIVE_STATUS);
-  const [sortAndFilter, setSortAndFilter] = useState<string[]>([])
-  const [filterLabels, setFilterLabels] = useState<string[]>(labels["ACTIVE_STATUS"]);
+  const classes = styles(props);
+  const { match } = props;
   const [searchValue, setSearchValue] = useState<string>("");
-  const fullCustomers: CustomerRecord[] = useSelector(selectAllCustomersArray) || [];
+  const [value, setValue] = React.useState('0');
+  const changeTab = (event: React.ChangeEvent<{}>, newValue: string) => {
+    setValue(newValue);
+  };
+
+  //default customer arrays
+  const allCustomers = useSelector(selectAllCustomersArray) || [];
+  const toMeterCustomers = useSelector(selectCustomersToMeter) || [];
+  const toCollectCustomers = useSelector(selectCustomersToCollect) || [];
+  const customersDone = useSelector(selectCustomersDone) || [];
+  const defaultCustomers: CustomerMenu = {
+    all: allCustomers,
+    toMeter: toMeterCustomers,
+    toCollect: toCollectCustomers,
+    done: customersDone,
+  }
 
+  //customer states
+  const [customers, setCustomers] = useState<CustomerMenu>(defaultCustomers);
+  const allCustomersTrie: TrieTree<CustomerRecord> = new TrieTree('name');
+  allCustomersTrie.addAll(allCustomers);
+
+  //default customer sets
+  const toMeterCustomersSet: Set<string> = new Set<string>(toMeterCustomers.map((customer: CustomerRecord) => customer.id));
+  const toCollectCustomersSet: Set<string> = new Set<string>(toCollectCustomers.map((customer: CustomerRecord) => customer.id));
+
+  //updates after every character typed
   useEffect(() => {
     getCustomers();
-    setSortAndFilter([SortBy[sortBy], FilterBy[filterBy]]);
-  }, [sortBy, filterBy, searchValue]);
+  }, [searchValue]);
 
-  const getCustomers = () => {
-    let allCustomers = fullCustomers;
+  const handleSearchChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const searchVal = e.target.value.trim();
+    setSearchValue(searchVal);
+  }
 
+  const getCustomers = () => {
     if (searchValue !== '') {
-      allCustomers = allCustomersTrie.get(searchValue);
+      const searchedCustomersArray = allCustomersTrie.get(searchValue);
+      const searchedCustomers = new Set<string>(searchedCustomersArray.map((customer: CustomerRecord) => customer.id));
+
+      const newCustomers: CustomerMenu = {
+        all: searchedCustomersArray,
+        toMeter: customers.toMeter.filter((customer: CustomerRecord) => searchedCustomers.has(customer.id)),
+        toCollect: customers.toCollect.filter((customer: CustomerRecord) => searchedCustomers.has(customer.id)),
+        done: customers.done.filter((customer: CustomerRecord) => searchedCustomers.has(customer.id)),
```

Comment:
Sorry I just remembered this, but the NPO wanted the search to be on both the `customerNumber` and `name` (so if you input a number it searches by customer number, if input a name it searches by name).